### ISSUE TITLE
Bug: Transitive import of @typescript-eslint/types from @typescript-eslint/type-utils

### ISSUE THREAD

**Issue by mrginglymus** [typescript-eslint/typescript-eslint#11382]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

type-utils

### Repro Code

```TypeScript
Error: @typescript-eslint/type-utils tried to access @typescript-eslint/types, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

Required package: @typescript-eslint/types (via "@typescript-eslint\types")
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

No transitive imports

### Actual Result

Transitive imports

### Versions

Latest

**Comment by mrginglymus** [comment]
Caused by https://github.com/typescript-eslint/typescript-eslint/commit/d1388fc0c9cd40dc3a32787e7ab04cb78bc904c9
